### DISCUSSION TITLE
ImageTagMirrorSet - Should they work with Builds?

### DISCUSSION THREAD

**Discussion by johnnysheppard-isode** [okd-project/okd#1632] [Q&A]
Hi,

I've been trying out the new ImageTagMirrorSet resource type from OKD 4.13. I can get it to work with pulling an image for a pod, but  it doesn't seem to work for a Build.

We run our OKD cluster without direct access to docker.io so we don't hit the download limits (blocked at the office firewall level). We use Artifactory as a mirror to cache pulls to Docker and it works fine internally with the Docker tools. It also works as I say, in a pod definition with the new ImageTagMirrorSet, but when I run a Build, it seems to timeout:

`Trying to pull docker.io/library/nginx:1.16.0-alpine...
time="2023-06-09T12:45:05Z" level=warning msg="Failed, retrying in 1s ... (1/3). Error: initializing source docker://nginx:1.16.0-alpine: pinging container registry registry-1.docker.io: Get \"https://registry-1.docker.io/v2/\": dial tcp 44.205.64.79:443: i/o timeout"
time="2023-06-09T12:46:06Z" level=warning msg="Failed, retrying in 2s ... (2/3). Error: initializing source docker://nginx:1.16.0-alpine: pinging container registry registry-1.docker.io: Get \"https://registry-1.docker.io/v2/\": dial tcp 34.205.13.154:443: i/o timeout"
time="2023-06-09T12:47:08Z" level=warning msg="Failed, retrying in 4s ... (3/3). Error: initializing source docker://nginx:1.16.0-alpine: pinging container registry registry-1.docker.io: Get \"https://registry-1.docker.io/v2/\": dial tcp 34.205.13.154:443: i/o timeout"`

Does anyone know, is there anything else I need to add, or should this be working?

**Comment by maps1995** [comment]
I have the same issue, it seems mirroring is not working in builds.
Do you have any idea @vrutkovs ? please help us.
